### PR TITLE
Added first quiz functionality with the Toast at the top

### PR DESCRIPTION
**Pull request by f7deleon**
Completed first chapter, added first quiz with the Toast at the top

### Related Trello cards:
https://trello.com/c/LfjGEJc4

### UI preview:
aw1ds3vthr

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    buildTypes {
+        release {
+            minifyEnabled false
+            proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
+        }
+    }
+}
+
+dependencies {
+    implementation fileTree(dir: 'libs', include: ['*.jar'])
```
Comment (mirland) [inline app/build.gradle]: sort dependencies

File: app/build.gradle [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    implementation"org.jetbrains.kotlin:kotlin-stdlib-jre7:$kotlin_version"
+    implementation 'com.android.support:appcompat-v7:28.0.0-alpha3'
```
Comment (mirland) [inline app/build.gradle]: you shouldn't use alpha versions

File: app/src/main/AndroidManifest.xml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+        <activity android:name=".QuizActivity">
+            <intent-filter>
+                <action android:name="android.intent.action.MAIN" />
+
+                <category android:name="android.intent.category.LAUNCHER" />
+            </intent-filter>
+        </activity>
+    </application>
+
+</manifest>
```
Comment (mirland) [inline app/src/main/AndroidManifest.xml]: add new line

File: app/src/main/AndroidManifest.xml [hunk 4]
```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="utf-8"?>
+<manifest xmlns:android="http://schemas.android.com/apk/res/android"
+    package="com.example.felipe.geoquiz">
+
+    <application
+        android:allowBackup="true"
+        android:icon="@mipmap/ic_launcher"
+        android:label="@string/app_name"
```
Comment (mirland) [inline app/src/main/AndroidManifest.xml:10]: indent file, use xl format

File: app/src/main/java/com/example/felipe/geoquiz/QuizActivity.kt [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_quiz)
+
+        trueButton =  findViewById(R.id.btnTrue)
+        trueButton.setOnClickListener { printResutl(R.string.correctToast)  }
+
+        falseButton = findViewById(R.id.btnFalse)
+        falseButton.setOnClickListener {printResutl(R.string.incorrectToast) }
+
```
Comment (mirland) [inline app/src/main/java/com/example/felipe/geoquiz/QuizActivity.kt]: remove it

File: app/src/main/java/com/example/felipe/geoquiz/QuizActivity.kt [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_quiz)
+
+        trueButton =  findViewById(R.id.btnTrue)
+        trueButton.setOnClickListener { printResutl(R.string.correctToast)  }
+
+        falseButton = findViewById(R.id.btnFalse)
+        falseButton.setOnClickListener {printResutl(R.string.incorrectToast) }
```
Comment (mirland) [inline app/src/main/java/com/example/felipe/geoquiz/QuizActivity.kt]: indentation is wrong

File: app/src/main/res/layout/activity_quiz.xml [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+            <Button
+                android:id="@+id/btnFalse"
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:text="@string/false_button" />
+        </LinearLayout>
+    </LinearLayout>
+
+
+</android.support.constraint.ConstraintLayout>
```
Comment (mirland) [inline app/src/main/res/layout/activity_quiz.xml]: nl

File: app/src/main/res/layout/activity_quiz.xml [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+                android:layout_height="wrap_content"
+                android:text="@string/true_button" />
+            <Button
+                android:id="@+id/btnFalse"
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:text="@string/false_button" />
+        </LinearLayout>
+    </LinearLayout>
+
```
Comment (mirland) [inline app/src/main/res/layout/activity_quiz.xml]: rm

File: app/src/main/res/mipmap-anydpi-v26/ic_launcher.xml [hunk 9]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
+<adaptive-icon xmlns:android="http://schemas.android.com/apk/res/android">
+    <background android:drawable="@drawable/ic_launcher_background" />
+    <foreground android:drawable="@drawable/ic_launcher_foreground" />
+</adaptive-icon>
```
Comment (mirland) [inline app/src/main/res/mipmap-anydpi-v26/ic_launcher.xml]: nl

File: app/src/main/res/mipmap-anydpi-v26/ic_launcher_round.xml [hunk 10]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
+<adaptive-icon xmlns:android="http://schemas.android.com/apk/res/android">
+    <background android:drawable="@drawable/ic_launcher_background" />
+    <foreground android:drawable="@drawable/ic_launcher_foreground" />
+</adaptive-icon>
```
Comment (mirland) [inline app/src/main/res/mipmap-anydpi-v26/ic_launcher_round.xml]: nl

File: app/src/main/res/values/colors.xml [hunk 11]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+    <color name="colorPrimary">#3F51B5</color>
```
Comment (mirland) [inline app/src/main/res/values/colors.xml]: use lower case

File: app/src/main/AndroidManifest.xml (hunk 3 above)
Comment (FerrariBruno) [inline app/src/main/AndroidManifest.xml]: new line at the end

File: app/src/main/res/values/strings.xml [hunk 12]
```diff
@@ -0,0 +1,8 @@
+<resources>
+    <string name="app_name">GeoQuiz</string>
+    <string name="true_button">True</string>
+    <string name="false_button">False</string>
+    <string name="question_text">Canberra is the capital of Australia.</string>
+    <string name="correctToast">Correct!</string>
```
Comment (mirland) [inline app/src/main/res/values/strings.xml]: the name is not good

File: app/src/main/res/values/strings.xml [hunk 13]
```diff
@@ -0,0 +1,8 @@
+<resources>
+    <string name="app_name">GeoQuiz</string>
+    <string name="true_button">True</string>
```
Comment (mirland) [inline app/src/main/res/values/strings.xml]: same, change the name

File: app/src/main/AndroidManifest.xml [hunk 14]
```diff
@@ -0,0 +1,21 @@
...
+        android:roundIcon="@mipmap/ic_launcher_round"
+        android:supportsRtl="true"
+        android:theme="@style/AppTheme">
+        <activity android:name=".QuizActivity">
```
Comment (FerrariBruno) [inline app/src/main/AndroidManifest.xml]: should have a line before

File: app/src/main/res/values/styles.xml [hunk 15]
```diff
@@ -0,0 +1,11 @@
+<resources>
+
+    <!-- Base application theme. -->
+    <style name="AppTheme" parent="Theme.AppCompat.Light.DarkActionBar">
+        <!-- Customize your theme here. -->
+        <item name="colorPrimary">@color/colorPrimary</item>
+        <item name="colorPrimaryDark">@color/colorPrimaryDark</item>
+        <item name="colorAccent">@color/colorAccent</item>
+    </style>
+
```
Comment (mirland) [inline app/src/main/res/values/styles.xml]: rm

File: app/src/main/res/values/styles.xml [hunk 16]
```diff
@@ -0,0 +1,11 @@
+<resources>
+
+    <!-- Base application theme. -->
```
Comment (mirland) [inline app/src/main/res/values/styles.xml]: rm

File: app/src/main/AndroidManifest.xml (hunk 14 above)
Comment (FerrariBruno) [reply app/src/main/AndroidManifest.xml]: and also put `android:name=".QuizActivity"` on the next line

File: app/src/main/AndroidManifest.xml [hunk 17]
```diff
@@ -0,0 +1,21 @@
...
+            <intent-filter>
```
Comment (FerrariBruno) [inline app/src/main/AndroidManifest.xml]: should have a new line before

File: gradle.properties [hunk 18]
```diff
@@ -0,0 +1,13 @@
+# Project-wide Gradle settings.
+# IDE (e.g. Android Studio) users:
+# Gradle settings configured through the IDE *will override*
+# any settings specified in this file.
+# For more details on how to configure your build environment visit
+# http://www.gradle.org/docs/current/userguide/build_environment.html
+# Specifies the JVM arguments used for the daemon process.
+# The setting is particularly useful for tweaking memory settings.
+org.gradle.jvmargs=-Xmx1536m
```
Comment (mirland) [inline gradle.properties]: avoid it, add it in your computer configuration file

**Review by mirland — CHANGES_REQUESTED** [review]
Indent all files

File: app/src/main/AndroidManifest.xml [hunk 19]
```diff
@@ -0,0 +1,21 @@
...
+                <action android:name="android.intent.action.MAIN" />
+
+                <category android:name="android.intent.category.LAUNCHER" />
+            </intent-filter>
```
Comment (FerrariBruno) [inline app/src/main/AndroidManifest.xml]: add a new line before and after

File: app/src/main/AndroidManifest.xml [hunk 20]
```diff
@@ -0,0 +1,21 @@
...
+        </activity>
```
Comment (FerrariBruno) [inline app/src/main/AndroidManifest.xml]: add a new line after

File: app/src/main/java/com/example/felipe/geoquiz/QuizActivity.kt [hunk 21]
```diff
@@ -0,0 +1,30 @@
...
+
+import android.support.v7.app.AppCompatActivity
+import android.os.Bundle
+import android.view.Gravity
+import android.widget.Button
+import android.widget.Toast
+import java.io.BufferedReader
+
+class QuizActivity : AppCompatActivity() {
+    private lateinit var trueButton:Button
```
Comment (FerrariBruno) [inline app/src/main/java/com/example/felipe/geoquiz/QuizActivity.kt]: leave a space after the `:`

File: app/src/main/java/com/example/felipe/geoquiz/QuizActivity.kt [hunk 22]
```diff
@@ -0,0 +1,30 @@
...
+    private lateinit var falseButton:Button
```
Comment (FerrariBruno) [inline app/src/main/java/com/example/felipe/geoquiz/QuizActivity.kt]: same

File: app/src/main/java/com/example/felipe/geoquiz/QuizActivity.kt [hunk 23]
```diff
@@ -0,0 +1,30 @@
...
+
+    fun printResutl(message: Int) =with (Toast.makeText(this ,message, Toast.LENGTH_SHORT)) {
```
Comment (FerrariBruno) [inline app/src/main/java/com/example/felipe/geoquiz/QuizActivity.kt]: mispelled function name, also add a space after `=`

File: app/src/main/res/layout/activity_quiz.xml [hunk 24]
```diff
@@ -0,0 +1,39 @@
...
+    android:layout_height="match_parent"
+    tools:context=".QuizActivity"
+    >
+
+    <LinearLayout
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"
+        android:gravity="center"
+        android:orientation="vertical">
+        <TextView
```
Comment (FerrariBruno) [inline app/src/main/res/layout/activity_quiz.xml]: leave a new line in between each of the elements in an `.xml`

File: app/src/main/res/layout/activity_quiz.xml [hunk 25]
```diff
@@ -0,0 +1,39 @@
...
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    tools:context=".QuizActivity"
+    >
+
+    <LinearLayout
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"
+        android:gravity="center"
+        android:orientation="vertical">
```
Comment (FerrariBruno) [inline app/src/main/res/layout/activity_quiz.xml]: always put the `>` in a separate (new) line, it's cleaner that way

Comment (FerrariBruno) [reply app/src/main/res/layout/activity_quiz.xml]: this applies to all of this file

File: app/src/main/res/layout/activity_quiz.xml (hunk 24 above)
Comment (FerrariBruno) [reply app/src/main/res/layout/activity_quiz.xml]: this applies to all of this file

File: app/src/main/AndroidManifest.xml (hunk 14 above)
Comment (FerrariBruno) [reply app/src/main/AndroidManifest.xml]: and the `>` in a separate (new) line too

File: app/src/main/AndroidManifest.xml [hunk 26]
```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="utf-8"?>
+<manifest xmlns:android="http://schemas.android.com/apk/res/android"
+    package="com.example.felipe.geoquiz">
```
Comment (FerrariBruno) [inline app/src/main/AndroidManifest.xml]: the > should be in a separate (new) line too

File: app/src/main/AndroidManifest.xml [hunk 27]
```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="utf-8"?>
+<manifest xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (FerrariBruno) [inline app/src/main/AndroidManifest.xml]: `xmlns:android="http://schemas.android.com/apk/res/android"` should be in a new line

File: app/src/main/java/com/example/felipe/geoquiz/QuizActivity.kt [hunk 28]
```diff
@@ -0,0 +1,30 @@
...
+        setGravity(Gravity.TOP,0,200)
+        show()
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
```
Comment (FerrariBruno) [inline app/src/main/java/com/example/felipe/geoquiz/QuizActivity.kt]: I think this method should be on top. It's the first method that's executed so it makes sense. In practice, all of the lifecycles methods go at the start of the class (after the variables declaration, of course)

File: app/src/main/java/com/example/felipe/geoquiz/QuizActivity.kt (hunk 5 above)
Comment (FerrariBruno) [inline app/src/main/java/com/example/felipe/geoquiz/QuizActivity.kt]: delete this blank line

**Review by FerrariBruno — CHANGES_REQUESTED** [review]

File: app/src/main/AndroidManifest.xml [hunk 29]
```diff
@@ -0,0 +1,35 @@
...
+        android:label="@string/app_name"
+        android:roundIcon="@mipmap/ic_launcher_round"
+        android:supportsRtl="true"
+        android:theme="@style/AppTheme"
+        >
+
+        <activity
+            android:name=".QuizActivity"
+            >
+            <intent-filter>
```
Comment (FerrariBruno) [inline app/src/main/AndroidManifest.xml]: you missed a line before this xD

File: app/build.gradle [hunk 30]
```diff
@@ -20,14 +20,18 @@ android {
...
+
+
 }
 
+final ANDROID_SUPPORT_VERSION = '27.1.1'
+
 dependencies {
+    androidTestImplementation 'com.android.support.test.espresso:espresso-core:3.0.2'
+    androidTestImplementation 'com.android.support.test:runner:1.0.2'
+    implementation "com.android.support:appcompat-v7:${ANDROID_SUPPORT_VERSION}"
```
Comment (mirland) [inline app/build.gradle]: maybe a new line here

File: app/build.gradle [hunk 31]
```diff
@@ -20,14 +20,18 @@ android {
...
+    implementation 'com.android.support.constraint:constraint-layout:1.1.2'
     implementation fileTree(dir: 'libs', include: ['*.jar'])
```
Comment (mirland) [inline app/build.gradle]: this is not necessary

File: app/src/main/java/com/example/felipe/geoquiz/QuizActivity.kt [hunk 32]
```diff
@@ -0,0 +1,31 @@
...
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_quiz)
+        trueButton =  findViewById(R.id.btnTrue)
+        trueButton.setOnClickListener { printResult(R.string.correct)  }
+
+        falseButton = findViewById(R.id.btnFalse)
+        falseButton.setOnClickListener {printResult(R.string.incorrect) }
+
```
Comment (FerrariBruno) [inline app/src/main/java/com/example/felipe/geoquiz/QuizActivity.kt]: remove this blank line

File: app/src/main/AndroidManifest.xml [hunk 33]
```diff
@@ -1,21 +1,35 @@
...
+            >
             <intent-filter>
-                <action android:name="android.intent.action.MAIN" />
 
-                <category android:name="android.intent.category.LAUNCHER" />
+                <action
+                    android:name="android.intent.action.MAIN"
+                    />
+
+                <category
```
Comment (mirland) [inline app/src/main/AndroidManifest.xml]: single liine

File: app/src/main/res/mipmap-anydpi-v26/ic_launcher.xml (hunk 9 above)
Comment (FerrariBruno) [reply app/src/main/res/mipmap-anydpi-v26/ic_launcher.xml]: new line

File: app/src/main/res/mipmap-anydpi-v26/ic_launcher_round.xml (hunk 10 above)
Comment (FerrariBruno) [reply app/src/main/res/mipmap-anydpi-v26/ic_launcher_round.xml]: same

**Review by FerrariBruno — APPROVED** [review]

File: app/src/main/java/com/example/felipe/geoquiz/QuizActivity.kt [hunk 34]
```diff
@@ -8,23 +8,24 @@ import android.widget.Toast
...
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_quiz)
-
         trueButton =  findViewById(R.id.btnTrue)
-        trueButton.setOnClickListener { printResutl(R.string.correctToast)  }
+        trueButton.setOnClickListener { printResult(R.string.correct)  }
 
         falseButton = findViewById(R.id.btnFalse)
-        falseButton.setOnClickListener {printResutl(R.string.incorrectToast) }
+        falseButton.setOnClickListener {printResult(R.string.incorrect) }
```
Comment (mirland) [inline app/src/main/java/com/example/felipe/geoquiz/QuizActivity.kt]: indentation

File: app/src/main/java/com/example/felipe/geoquiz/QuizActivity.kt [hunk 35]
```diff
@@ -8,23 +8,24 @@ import android.widget.Toast
...
 
     }
+
+    fun printResult(message: Int) = with (Toast.makeText(this ,message, Toast.LENGTH_SHORT)) {
+        setGravity(Gravity.TOP,0,200)
+        show()
+    }
+
```
Comment (mirland) [inline app/src/main/java/com/example/felipe/geoquiz/QuizActivity.kt]: rm

File: app/build.gradle [hunk 36]
```diff
@@ -0,0 +1,38 @@
+apply plugin: 'com.android.application'
+
+apply plugin: 'kotlin-android'
+
```
Comment (mirland) [inline app/build.gradle:4]: rm

File: app/src/main/java/com/example/felipe/geoquiz/Question.kt [hunk 37]
```diff
@@ -0,0 +1,14 @@
...
+
+constructor( textResId:Int,  answerTrue:Boolean)
+
+init {
+    this.textResId = textResId
+    this.answerTrue = answerTrue
+}
+
+public textResId:Int
+}
```
Comment (mirland) [inline app/src/main/java/com/example/felipe/geoquiz/Question.kt:14]: nl

File: app/src/main/java/com/example/felipe/geoquiz/Question.kt [hunk 38]
```diff
@@ -0,0 +1,14 @@
+package com.example.felipe.geoquiz
+
+private val textResId:Int
```
Comment (mirland) [inline app/src/main/java/com/example/felipe/geoquiz/Question.kt:3]: ?

**Review by mirland — APPROVED** [review]

**Pull request merged**